utils/get_supported_scylla_base_versions.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+import os
+
+from pkg_resources import parse_version
+from sdcm.utils.version_utils import is_enterprise
+from sdcm.utils.common import get_s3_scylla_repos_mapping
+
+sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
+
+
+# We support to migrate from specific OOS version to enterprise
+supported_src_oos = {'2021.1': '4.3', '2020.1': '4.0', '2019.1': '3.0'}
+# If new support distro shared repo with others, we need to assign the start support versions. eg: centos8
+start_support_versions = {'centos-8': {'scylla': '4.1', 'enterprise': '2021.1'}}
+
+
+class UpgradeBaseVersion:  # pylint: disable=too-many-instance-attributes
+
+    oos_start_support_version = None
+    ent_start_support_version = None
+
+    def __init__(self, scylla_repo: str, linux_distro: str, scylla_version: str = None):
+        if len(linux_distro.split('-')) > 1:
+            self.dist_type, *_, self.dist_version = linux_distro.split('-')
+        else:
+            self.dist_type = linux_distro.split('-')[0]
+            self.dist_version = None
+        self.scylla_repo = scylla_repo
+        self.linux_distro = linux_distro
+        self.product, self.scylla_version = self.get_product_and_version(scylla_version)
+        self.repo_maps = get_s3_scylla_repos_mapping(self.dist_type, self.dist_version)
+
+    def get_product_and_version(self, scylla_version: str = None):
+        """
+        return scylla product name and major version. if scylla_version isn't assigned,
+        we will try to get the major version from the scylla_repo url.
+        """
+        if scylla_version is None:
+            assert 'unstable/' in self.scylla_repo
```

Comment:
What happens when we try to upgrade from AMI?

Reply:
The above self.scylla_repo is the new_scylla_repo parameter, AMI upgrade job also has it.

The correct base_versions list will be generated,  each base_version will be set to scylla_version parameter, and then ami_id_db_scylla will be generated.

Reply:
We always set base_version to scylla_version in upgrade pipeline, the scylla_version will be converted in sdcm/sct_config.py: (line 1393         # 6) handle scylla_version if exists)

- AMI: scylla_version -> ami_id_db_scylla
- non-AMI:  scylla_version -> scylla_repo

```
vars/rollingOperatorUpgradePipeline.groovy
123             stage('Run SCT stages') {
124                 steps {
125                     script {
126                         def tasks = [:]
127                         def base_versions_list = readJSON text: params.base_versions
128                         for (base_version in base_versions_list) {
129                             run_params =  readJSON text: groovy.json.JsonOutput.toJson(params)
130                             run_params.scylla_version = base_version  <<<<<<<<
131                             tasks["Scylla Operator upgrade from ${base_version} to ${new_version}"] = {
132 
133                                 stage("${base_version} -> ${new_version} - Running test") {
134                                     catchError(stageResult: 'FAILURE') {
135                                         script {
136                                             wrap([$class: 'BuildUser']) {
137                                                 dir('scylla-cluster-tests') {
138                                                     timeout(time: testRunTimeout, unit: 'MINUTES') {
139                                                         checkout scm
140                                                         runSctTest(run_params, builder.region)
141                                                     }
142                                                 }
143                                             }
144                                         }
145                                     }
146                                 }
```